patrick-lab/server.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Client = require('./model/pages.js');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = net.createServer();
+const PORT = process.env.PORT || 3000;
+
+const pool = [];
+
+ee.on('default', (client, string) => {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('/all', (client, string) => {
+  pool.forEach(c => c.socket.write(`${client.nickname}: ${string}`));
+});
+
+ee.on('/nick', (client, string) => {
+  client.nickname = string.trim();
+});
+
+ee.on('/dm', (client, string) => {
+  let target = string.split(' ')[0];
+  let message = string.split(' ').slice(1).join('');
+  pool.forEach(c => {
+    if(c.nickname === target) {
```

Comment:
**TODO FOR LOST POINTS**

Line 32 is not actually happening because you actually need to add a ``.trim()`` to line 25, i.e.:

```
let target = string.split(' ')[0].trim();
```